components/base-war/src/main/java/org/xwiki/velocity/tools/JSONTool.java

```diff
@@ -127,6 +127,46 @@ public JSON parse(String json)
         }
     }
 
+    /**
+     * Parse a serialized JSON into a real JSONObject. Only valid strings can be parsed, and doesn't support
+     * JSONP. If the argument is not valid JSON, then {@code null} is returned.
+     *
+     * @param json the string to parse, must be valid JSON
+     * @return the parsed JSON, that is a either a {@link org.json.JSONObject}, or {@code null} if the argument is not
+     * a valid JSON
+     * @since 1.3M4
+     */
+    public JSONObject parseToJSONObject(String json)
+    {
+        try {
+            return new JSONObject(json);
+        } catch (org.json.JSONException ex) {
+            this.logger.info("Tried to parse invalid JSON: [{}], exception was: {}",
+                    StringUtils.abbreviate(json, 32), ex.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * Parse a serialized JSON into a real JSONArray. Only valid strings can be parsed, and doesn't support
+     * JSONP. If the argument is not valid JSON, then {@code null} is returned.
```

Comment:
@allasm Something like "[1, 2, 2]" should work, but yes, the second one will throw an exception.